### DISCUSSION TITLE
dx11 grabber introduces "stuttering" in video games

### DISCUSSION THREAD

**Discussion by pho084** [awawa-dev/HyperHDR#173] [General]
Hi,

this is a problem i have already experienced with hyperionscreencapture but it seems to persist here as well.

whenever the hyperhdr dx11 screen capture is activated games start to stutter very subtly but annoyingly in high demanding areas. this is not performance related as i can turn down details/resolution and monitor GPU load. 

However that's exactly what it feels like: the problem seems to me like a refresh rate problem; it is that slight stutter that you usually experience when the gpu and TV's refresh rate don't exactly match. but even vsynch does not fix the problem in this case.

could it be the mismatch between 60hz screen refresh rate and the refresh rate of the hyperhdr capture? I tried upping the fps in capture settings but unfortunately i cannot increase them beyond 30fps. it would be great for testing purposes if future versions could support 60fps capturing. Because even 30fps (i.e. half of the screens refresh rate) does not improve the stuttering

what's weird is that it is not apparent all the time, some areas of a game will run fluidly while others produce this stuttering effect.
turning off HyperHDR fixes the stuttering ingame.

i will do some more testing, also with HSC, but any ideas would be much appreciated.

**Comment by awawa-dev** [comment]
Hi
As I wrote in the blog post when v17 was released, using software grabbers are risky for GPU acceleration. That shuttering could be expected, I suspect locking resources needed to acquire a texture surface with a captured frame disrupts the game video processing/rendering and it's not related to CPU/GPU load: it's an access problem. 

Software screen grabber's rate in HyperHDR is completely unrelated with the screen refresh rate...consider it as just a timer with some interval when HyperHDR starts to capture a frame. But to do that it must lock DX surface for a while.

**Comment by pho084** [comment]
thank you for the reply!
so if i understand you correctly there is not much hope in fixing this apart from going back to usb capture. that in turn is not an option for me in this case as i will lose VRR/Gsynch support by splitting the hdmi signal :(

so my hope lies in piccap providing the capture through WebOS i.e. the TV itself (which seems not without risk). or the webcam solution (which seems far from optimal)

**Comment by pho084** [comment]
i thihnk i found a solution:

if your TV/screen has Gsynch/VRR compatibility try the following: Set refresh rate to maximum (i.e. 120 for OLED tvs), unlock max-fps ingame, disable vsynch ingame, run HyperHDR. If Gsynch works correctly it should compensate for pretty much all the stuttering.

works like a charm for me at least (turns out i had gsynch improperly configured all along)
